src/build/arg/mod.rs

```diff
@@ -269,6 +269,10 @@ impl<'help> Arg<'help> {
     /// ```
     /// [`short`]: ./struct.Arg.html#method.short
     pub fn short(mut self, s: char) -> Self {
+        if s == '-' {
+            panic!("short option name cannot be `-`");
```

Comment:
@CreepySkeleton All these panics can be made compile time in clap_derive, right? So, we should think about a way to store all these panics in one place as a single source of truth.

Reply:
Some of them - yes, definitely, but not all. I'm not familiar enough with these parts of codebase yet so let's postpone this discussion until we get to bugfixing. I'll get the import ready by tomorrow, it's 70% complete.